client/src/planwise/client/sources/components/dropdown.cljs

```diff
@@ -14,29 +15,17 @@
    [(subscribe [:sources/list])
     (subscribe [:projects2/source-types])])
  (fn [[list types] _]
-   (->> list
-        (filter (fn [{:keys [type]}] (types type)))
-        (mapv (fn [{:keys [id name type]}] {:value id :label name :type type})))))
+   (filter (fn [{:keys [type]}] (types type)) list)))
 
 ;; ----------------------------------------------------------------------------
 ;; Views
 
-(defn- sources-select-component
-  [{:keys [label value options empty-label on-change]}]
-  [m/Select {:label (if (empty? options) empty-label label)
-             :disabled (empty? options)
-             :value (str value)
-             :options options
-             :onChange #(on-change (js/parseInt (-> % .-target .-value)))}])
-
 (defn sources-dropdown-component
-  [{:keys [label value on-change disabled?]}]
-  (let [options   (subscribe [:sources/dropdown-options])
-        component (if (or disabled? (empty? @options))
-                    common2/disabled-input-component
-                    sources-select-component)]
-    [component {:label        label
-                :value        value
-                :options      @options
-                :empty-label  "No sources layer available."
-                :on-change    on-change}]))
\ No newline at end of file
+  [attrs]
+  (let [props (merge {:choices   @(rf/subscribe [:sources/dropdown-options])
+                      :label-fn  :name
+                      :render-fn (fn [source] [:div.option-row
+                                               [:span (:name source)]
+                                               [:span.option-context (:value source)]])}
+                     attrs)]
+    (into [filter-select/single-dropdown] (mapcat identity props))))
```

Comment:
This view is essentially the same as the one for the `regions-dropdown-component`, so I guess there's a component we should extract - but I didn't trust my ClojureScript-foo enough to go for it now.